File: OC/XWMP3ToWAV/XWMP3ToWAV/ViewController.m
//
//  ViewController.m
//  XWMP3ToWAV
//
//  Created by 邱学伟 on 2017/3/13.
//  Copyright © 2017年 邱学伟. All rights reserved.
//

#import "ViewController.h"
#import "XWMP3ToPCM.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self testMP3ToPCM];
}

- (void)testMP3ToPCM {
    
    NSString *mp3FilePath = [[NSBundle mainBundle] pathForResource:@"libai.mp3" ofType:nil];
    NSString *pcmFilePath = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES).lastObject;
    pcmFilePath = [pcmFilePath stringByAppendingPathComponent:@"libai.mp3.wav"];
    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
        [XWMP3ToPCM mp3ToPcmWithMp3FilePath:mp3FilePath pcmFilePath:pcmFilePath];
    });
}


@end

File: Swift/XWMP3ToWAV/XWMP3ToWAV/ViewController.swift
//
//  ViewController.swift
//  XWMP3ToWAV
//
//  Created by 邱学伟 on 2017/3/13.
//  Copyright © 2017年 邱学伟. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
       
        self.textMP3ToPCM()
    }
}

extension ViewController {
    
    func textMP3ToPCM() {
        guard let mp3FilePath : String = Bundle.main.path(forResource: "NoGoodbye.mp3", ofType: nil) else { print("mp3 资源文建为空"); return }
        let docPath : String = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true).last!
        // 这里的扩展名'.wav'只是标记了文件的打开方式，实际的编码封装格式由assetWriter的fileType决定
        let pcmFilePath = docPath + "/NoGoodbye.mp3.wav"
        DispatchQueue.global().async {
            XWMP3ToPCM.mp3ToPcm(mp3FilePath: mp3FilePath, pcmFilePath: pcmFilePath)
        }
    }
}

